airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/job.py

```diff
@@ -245,19 +270,36 @@ def _on_created_job(self, **kwargs) -> None:
     def _on_canceled_job(self, response: requests.Response) -> Optional[AirbyteTracedException]:
         if not self._job_self_canceled:
             raise ShopifyBulkExceptions.BulkJobCanceled(
-                f"The BULK Job: `{self._job_id}` exited with {self._job_state}, details: {response.text}",
+                f"The BULK Job: `{self._job_id}` exited with {self._job_state}, details: {response.text}"
             )
+        else:
+            if self._job_should_checkpoint:
+                # set the flag to adjust the next slice from the checkpointed cursor value
+                self._job_adjust_slice_from_checkpoint = True
+                # fetch the collected records from CANCELED Job on checkpointing
+                self._job_result_filename = self._job_get_result(response)
 
     def _on_canceling_job(self, **kwargs) -> None:
         sleep(self._job_check_interval)
 
+    def _cancel_on_long_running_job(self) -> None:
+        self.logger.info(
+            f"Stream: `{self.http_client._name}` the BULK Job: {self._job_id} runs longer than expected ({self._job_max_elapsed_time} sec). Retry with the reduced `Slice Size` after self-cancelation."
+        )
+        self._job_long_running_cancelation = True
```

Comment:
We set this field to True, but as far as I can tell, we don't actually read or use this value. Do we still need it outside of assertions on tests? And if this is purely to validate tests, let's comment in the above field declaration that it's included for tests

Reply:
Removed! That was a leftover from couple of previous test attempts. Thanks!